kdl-client/src/Hooks/useProjectNavigation.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMemo } from 'react';
+import ROUTE, { buildRoute } from '../Constants/routes';
+import IconHome from '@material-ui/icons/Dashboard';
+import IconSettings from '@material-ui/icons/Settings';
+import { OverridableComponent } from '@material-ui/core/OverridableComponent';
+import { SvgIconTypeMap } from '@material-ui/core';
+
+export interface routeConfiguration {
+  label: string;
+  icon: OverridableComponent<SvgIconTypeMap<{}, 'svg'>>;
+}
+interface projectRouteConfiguration {
+  [key: string]: routeConfiguration;
+}
+
+export const projectRoutesConfiguration: projectRouteConfiguration = {
```

Comment:
interface here is not necessary as the keys are unknown, you can type this with object typing style.

```suggestion
export const projectRoutesConfiguration: { [key: ROUTE]: RouteConfigurration } = {
```